OpenRA.Mods.Common/Traits/World/SmudgeLayer.cs

```diff
@@ -163,14 +163,32 @@ public void AddSmudge(CPos loc)
 			}
 		}
 
+		public void RemoveSmudge(CPos loc)
+		{
+			if (dirty.ContainsKey(loc))
+			{
+				var tile = dirty[loc];
+				tile.Sprite = null;
+				dirty[loc] = tile;
+			}
+			else
+			{
+				var st = smudges.Keys.Random(world.SharedRandom);
+				dirty[loc] = new Smudge { Type = st, Depth = 0, Sprite = null };
```

Comment:
You should be able to simplify this to @RoosterDragon's suggestion now then, right?